cmd/state/main.go

```diff
@@ -207,6 +207,13 @@ func run(args []string, isInteractive bool, cfg *config.Instance, out output.Out
 
 	// Set up conditional, which accesses a lot of primer data
 	sshell := subshell.New(cfg)
+	if isInteractive {
+		// Disable terminal echo while State Tool is running.
+		// Other than in prompts and subshells (which temporarily re-enable echo), user typing should
+		// not interfere with output (e.g. runtime progress bars).
+		sshell.TurnOffEcho()
+		defer sshell.TurnOnEcho()
```

Comment:
Does this also disable ctrl+c and other similar inputs? I don't believe that's something we want to disallow.

Reply:
Also, is it possible that if the state tool errors mid-execution, we could leave the user's shell in a state where they cannot echo?

Reply:
You didn't see this, but in my original PR (https://github.com/ActiveState/cli/pull/2768), I noted that echo is restored after ^C, so no problem there. The `defer` does its job.

If there is a hard crash, echo will probably be lost, but this is the case for any terminal program that messes with termios, so the only resolution is for the user to exit this terminal and start up a new one.

Reply:
It looks like the `defer` will run even on a panic as long as it's on the function call stack so we should be good even in the event of a crash as this defer is placed ahead of the actual command logic.